helpers/network-recorder.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* global WebInspector:false */
+
+// Required for a select portion DevTools frontend to work (in node)
+global.self = global;
+global.Protocol = {
+  Agents() {}
+};
+global.WebInspector = {
+  _moduleSettings: {
+    cacheDisabled: {
+      addChangeListener() {},
+      get() {
+        return false;
+      }
+    },
+    monitoringXHREnabled: {
+      addChangeListener() {},
+      get() {
+        return false;
+      }
+    }
+  },
+  moduleSetting: function(settingName) {
+    return this._moduleSettings[settingName];
+  }
+};
+// Enum from chromium/src/out/Debug/gen/devtools/inspector.js
```

Comment:
Enum is from chromium//src/third_party/WebKit/Source/core/loader/MixedContentChecker.h

and since it's a security concern it's defined there and not redefined in JS really.

so this appears to be fine. cc @lgarron in case he recommends a different way